core/capabilities/ccip/launcher/deployment.go

```diff
@@ -0,0 +1,131 @@
+package launcher
+
+import (
+	"fmt"
+
+	cctypes "github.com/smartcontractkit/chainlink/v2/core/capabilities/ccip/types"
+
+	"go.uber.org/multierr"
+
+	ccipreaderpkg "github.com/smartcontractkit/chainlink-ccip/pkg/reader"
+)
+
+// activeCandidateDeployment represents a active-candidate deployment of OCR instances.
+type activeCandidateDeployment struct {
+	// active is the active OCR instance.
+	// active must always be present.
+	active cctypes.CCIPOracle
+
+	// candidate is the candidate OCR instance.
+	// candidate may or may not be present.
+	// candidate must never be present if active is not present.
+	candidate cctypes.CCIPOracle
+}
+
+// ccipDeployment represents active-candidate deployments of both commit and exec
+// OCR instances.
+type ccipDeployment struct {
+	commit activeCandidateDeployment
+	exec   activeCandidateDeployment
+}
+
+// Close shuts down all OCR instances in the deployment.
+func (c *ccipDeployment) Close() error {
+	var err error
+
+	// shutdown active commit instance.
+	err = multierr.Append(err, c.commit.active.Close())
```

Comment:
From the other PR ;) 
<img width="811" alt="image" src="https://github.com/user-attachments/assets/da09c3a4-a965-4695-b3a4-de7f27e71204">

Reply:
Nice, didn't know they are recommending stdlib. It's my preference also -- but in a separate PR from the renaming please :)
```
var errs []error

errs = append(errs, c.commit.active.Close())

...

return errors.Join(errs...)
```